amanuensis/blueprints/filterset.py

```diff
@@ -1,53 +1,21 @@
 import flask
-
-# from amanuensis.auth import login_required, current_token
-from amanuensis.resources.filterset import (
-    get_all,
-    get_by_id,
-    create,
-    delete,
-    update,
-    create_snapshot,
-    get_snapshot,
-)
-from amanuensis.config import config
+from amanuensis.resources.userdatamodel.search import get_filter_sets, create_filter_set, update_filter_set
+from amanuensis.resources.userdatamodel.search_is_shared import get_shared_filter_sets, create_filter_set_snapshot
 from amanuensis.auth.auth import current_user
 from amanuensis.errors import AuthError, UserError
-from amanuensis.schema import SearchSchema
 from cdislogging import get_logger
+from amanuensis.schema import SearchSchema, SearchIsSharedSchema
 
 
 logger = get_logger(__name__)
 
 blueprint = flask.Blueprint("filter-sets", __name__)
 
 
-@blueprint.route("/", methods=["GET"])
-# @login_required({"user"})
-def get_filter_sets():
-    try:
-        logged_user_id = current_user.id
-    except AuthError:
-        logger.warning("Unable to load or find the user, check your token")
-
-    # get the explorer_id from the querystring
-    explorer_id = flask.request.args.get("explorerId", default=1, type=int)
-
-    filter_sets = [
-        {
-            "name": s.name,
-            "id": s.id,
-            "description": s.description,
-            "filters": s.filter_object,
-            "ids": s.ids_list,
-        }
-        for s in get_all(logged_user_id, explorer_id)
-    ]
-    return flask.jsonify({"filter_sets": filter_sets})
-
 
+@blueprint.route("/", methods=["GET"])
 @blueprint.route("/<filter_set_id>", methods=["GET"])
-def get_filter_set(filter_set_id):
+def get_filter_set(filter_set_id=None):
```

Comment:
what does the option of None do here?

Reply:
It allows someone to send a GET request to filter-sets/ or filter-sets/<filter_set_id> but use the same function. So the None just avoids python throwing an error if no parameter is passed